helper/src/main/java/io/github/libxposed/helpers/BaseFinder.java

```diff
@@ -0,0 +1,254 @@
+     *
+     * @return this
+     */
+    public final S filterNonProtected() {
+        return filter(ModifierHelper::isNotProtected);
+    }
+
+    /**
+     * Filter methods/constructors if they have private modifier.
+     *
+     * @return this
+     */
+    public final S filterPrivate() {
+        return filter(ModifierHelper::isPrivate);
+    }
+
+    /**
+     * Filter methods/constructors if they not have private modifier.
+     *
+     * @return this
+     */
+    public final S filterNonPrivate() {
+        return filter(ModifierHelper::isNotPrivate);
+    }
+
+    /**
+     * Filter methods/constructors if they have native modifier.
+     *
+     * @return this
+     */
+    public final S filterNative() {
+        return filter(ModifierHelper::isNative);
+    }
+
+    /**
+     * Filter methods/constructors if they have native modifier.
+     *
+     * @return this
+     */
+    public final S filterNonNative() {
+        return filter(ModifierHelper::isNotNative);
+    }
+
+    /**
+     * Filter methods/constructors if they are package private.
+     *
+     * @return this
+     */
+    public final S filterPackage() {
+        return filter(ModifierHelper::isPackage);
+    }
+
+    /**
+     * Filter methods/constructors if they are not package private.
+     *
+     * @return this
+     */
+    public final S filterNonPackage() {
+        return filter(ModifierHelper::isNotPackage);
+    }
+
+    /**
+     * Filter methods/constructors by their exception types(Full-matches, cannot be null)
+     *
+     * @param e exception types
+     * @return this
+     */
+    public final S filterByExceptionType(Class<?>... e) {
+        var set1 = Set.of(e);
+
+        return filter(m -> {
+            var set2 = Set.of(getExceptionTypes(m));
+            return set1.equals(set2);
+        });
+    }
+
+    /**
+     * Filter methods/constructors by parameter types, make sure length is same as target method parameter types length.
+     *
+     * @param parameterTypes parameter types, use null to skip check some parameters.
+     * @return this
+     */
+    public final S filterByParameterTypes(Class<?>... parameterTypes) {
+        return filter(member -> {
+            var paramTypes = getParameterTypes(member);
+
+            if (paramTypes.length != parameterTypes.length) {
+                return false;
+            }
+            for (int i = 0; i < paramTypes.length; i++) {
+                // ignore if null
+                if (parameterTypes[i] == null)
+                    continue;
+
+                if (!paramTypes[i].equals(parameterTypes[i])) {
+                    return false;
+                }
+            }
+            return true;
+        });
+    }
+
+    /**
+     * Filter methods/constructors by parameter count.
+     *
+     * @param parameterCount parameter count
+     * @return this
+     */
+    public final S filterByParameterCount(int parameterCount) {
```

Comment:
What if I need to filter methods with parameter counts in range? consider filter by a predicate.